Realm/RLMRealm.h

```diff
@@ -500,6 +505,96 @@ typedef void (^RLMNotificationBlock)(RLMNotification notification, RLMRealm *rea
+ Synchronous try/catch handling will not work with the asynchronous transactions.
+ Asynchronous exceptions will be dispatched to the handler as `NSErrors`.
+ @note pass `nil` as a parameter to remove the handler
+ */
+- (void)setAsyncErrorHandler:(nullable RLMRealmAsyncErrorHandler)block;
+
+/**
+ Begins asynchronous write transaction.
+ 
+ @param block The block containing actions to perform.
+        `block` should end by calling `commitAsyncWriteTransaction`, `cancelAsyncTransaction:`,
+        `commitWriteTransaction` or `cancelWriteTransaction`.
+        Returning without one of these calls will be equivalent to calling `cancelAsyncTransaction`.
+ 
+ @return The Id for the asynchronous transaction.
+ 
+ @note `block` is queued for execution on the scheduler associated with the current realm.
+       It will run after the write mutex has been acquired.
+       The call returns immediately allowing the caller to proceed while the write mutex is held by someone else.
+       Write blocks from multiple calls to `beginAsyncWriteTransaction:` or `asyncTransactionWithBlock:`
+       will be executed in order.
+       A later call to `beginAsyncWriteTransaction:` or `asyncTransactionWithBlock:` will wait for any earlier
+       write blocks.
+ */
+- (RLMAsyncTransactionId)beginAsyncWriteTransaction:(void(^)())block;
+
+/** Commit asynchronous transaction.
+ 
+ The call returns immediately allowing the caller to proceed while the I/O is performed
+ on a dedicated background thread.
+ 
+ @param completionBlock  is queued for execution on the scheduler associated with
+        the current realm. It will run after the commit has reached stable storage.
+        Callbacks to `completionBlock` will occur in the order of `commitAsyncWriteTransaction`
+ 
+ @param isGroupingAllowed If `true`, the next `commitAsyncWriteTransaction` *may* run without an
+        intervening synchronization of stable storage.  Such a sequence of commits
+        form a group. In case of a platform crash, either none or all of the commits
+        in a group will reach stable storage. `false` by default.
+
+ @return Asynchronous transaction's Id.
+*/
+- (RLMAsyncTransactionId)commitAsyncWriteTransaction:(nullable void(^)())completionBlock isGroupingAllowed:(BOOL)isGroupingAllowed;
+/// :nodoc:
+- (RLMAsyncTransactionId)commitAsyncWriteTransaction:(void(^)())completionBlock;
+/// :nodoc:
+- (RLMAsyncTransactionId)commitAsyncWriteTransaction;
+
+/** Cancel a queued code block (either for an `asyncTransactionWithBlock`
+ or for an `commitAsyncWriteTransaction`)
+ 
+ @note Cancelling a commit will not abort the commit, it will only cancel the callback
+       informing of commit completion.
+ 
+ @param RLMAsyncTransactionId The transaction Id returned by `beginAsyncWriteTransaction` or  `commitAsyncWriteTransaction`
+*/
+- (void)cancelAsyncTransaction:(RLMAsyncTransactionId)asyncTransactionId;
+
+/** Commits a code block to asynchronous transactions queue.
```

Comment:
Using the word "commit" here is confusing because it's a very different meaning of the word from what we're usually talking about when we use "commit" related to a write.